### PR TITLE
Updates ssh connhelper error messages

### PR DESCRIPTION
**Pull request by arcenik**
Signed-off-by: François Scala <arcenik@github.com>

Here is a small PR that updates the error messages of ssh connection helper to make tham more clear/explicit.

### CODE REVIEW

**Comment by SvenDowideit** [comment]
I like it, but... :)

if a user gets an error containing the string `connhelper`, they should be able to search the docs for it too

https://docs.docker.com/search/?q=connhelper

gets me no results, but that could also be because i'm not familiar with how the docker docs search works these days.

luckily, its not been my call for many many years - @thaJeztah :)

**Comment by arcenik** [comment]
@SvenDowideit I think the opposite. The error message must be descriptive and explicit. When the error condition is too complex to be explained in one sentence, it should give an error code that lead to a documentation article with more details.

**Comment by SvenDowideit** [comment]
@arcenik I agree that error messages should be specific and lead to documentation - that's my point, your PR needs some changes to the docs so that searching the documentation results in a hit - which right now, it appears not to.

File: cli/connhelper/ssh/ssh.go [hunk 1]
```diff
@@ -23,30 +23,30 @@ func parseSSHURL(daemonURL string) (*sshSpec, error) {
 		return nil, err
 	}
 	if u.Scheme != "ssh" {
-		return nil, errors.Errorf("expected scheme ssh, got %s", u.Scheme)
+		return nil, errors.Errorf("ssh connhelper expected scheme ssh, got %s", u.Scheme)
```
Comment (thaJeztah) [inline cli/connhelper/ssh/ssh.go]: Perhaps use `%q` instead of `%s` so that the actual value is quotes (clearer to see what the problematic value is)

Comment (arcenik) [reply cli/connhelper/ssh/ssh.go]: done

**Pull request merged**